gcloud-java-storage/src/main/java/com/google/gcloud/storage/Blob.java

```diff
@@ -256,4 +261,88 @@ public URL signUrl(long expirationTimeInSeconds, SignUrlOption... options) {
   public Storage storage() {
     return storage;
   }
+
+  /**
+   * Gets the requested blobs. If {@code infos.length == 0} an empty list is returned. If
+   * {@code infos.length > 1} a batch request is used to fetch blobs.
+   *
+   * @param storage the storage service used to issue the request
+   * @param infos the blobs to get
+   * @return a list of {@code Blob} objects. If a blob does not exist the corresponding item in the
+   * list is {@code null}.
+   */
+  public static List<Blob> get(final Storage storage, BlobInfo... infos) {
+    checkNotNull(storage);
+    checkNotNull(infos);
+    int length = infos.length;
+    switch (length) {
+      case 0:
+        return Collections.emptyList();
+      case 1:
+        return Collections.singletonList(
+            new Blob(storage, storage.get(infos[0].bucket(), infos[0].name())));
+      default:
+        return Lists.transform(
+            storage.get(infos[0], infos[1], Arrays.copyOfRange(infos, 2, length)),
+            new Function<BlobInfo, Blob>() {
+              @Override
+              public Blob apply(BlobInfo f) {
+                return f != null ? new Blob(storage, f) : null;
+              }
+            });
+    }
+  }
+
+  /**
+   * Updates the requested blobs. If {@code infos.length == 0} an empty list is returned. If
+   * {@code infos.length > 1} a batch request is used to update blobs.
+   *
+   * @param storage the storage service used to issue the request
+   * @param infos the blobs to update
+   * @return a list of {@code Blob} objects. If a blob does not exist the corresponding item in the
+   * list is {@code null}.
+   */
+  public static List<Blob> update(final Storage storage, BlobInfo... infos) {
+    checkNotNull(storage);
+    checkNotNull(infos);
+    int length = infos.length;
+    switch (length) {
+      case 0:
+        return Collections.emptyList();
```

Comment:
Right, though if we think user is typically going to have collections (rather than just using
the varargs directly) then maybe we should add or replace with `Iterable<BlobInfo>`?

I am fine as is, and if we change would rather have it consistent.